### PR TITLE
Bluetooth: Controller: Fix CIS offset_min for dissimilar interval

### PR DESCRIPTION
**Pull request by cvinayak**
Fix CIS offset_min calculation to consider that ACL connection
interval and ISO intervals can be dissimilar and the offset
value has to be compensated for the latency until the
instant at which the offset is used.

Fix Central CIS SN and NESN for skipped events during CIS
setup. It is possible that CIG event may overlap the ACL or
other events at the CIS create instant, and hence have non
zero lazy value during the CIS setup events until CIS is
established.

Fix SN value on Peripheral CIS event abort. Adjust the SN
value based on the burst number such that the ISO data where
transmitted as normal.

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_peripheral_iso.c [hunk 1]
```diff
@@ -403,6 +403,21 @@ static void abort_cb(struct lll_prepare_param *prepare_param, void *param)
...
 
+		/* FIXME: Consider Flush Timeout when resetting current burst number */
+		if (!has_tx) {
+			has_tx = 1U;
+
+			/* Adjust nesn when flushing Tx */
+			/* FIXME: When Flush Timeout is implemented */
+			if (bn_tx <= cis_lll->tx.bn) {
+				/* sn and nesn are 1-bit, only Least Significant bit is needed */
+				cis_lll->sn += cis_lll->tx.bn + 1U - bn_tx;
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_peripheral_iso.c:414]: Can you elaborate on the `+ 1U` here?

File: subsys/bluetooth/controller/ll_sw/ull_central_iso.c [hunk 2]
```diff
@@ -953,6 +955,28 @@ static void mfy_cis_offset_get(void *param)
 			cig_remainder_us + cig->sync_delay -
 			acl_remainder_us - cis->sync_delay;
 
+	/* Calculate instant latency */
+	latency_acl = cis->central.instant - ull_conn_event_counter(conn);
+	elapsed_acl_us = latency_acl * conn->lll.interval * CONN_INT_UNIT_US;
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/ull_central_iso.c:960]: Is 32-bit always enough to cover the result of `latency_acl * conn->lll.interval * CONN_INT_UNIT_US`?

File: subsys/bluetooth/controller/ll_sw/ull_central_iso.c [hunk 3]
```diff
@@ -953,6 +955,28 @@ static void mfy_cis_offset_get(void *param)
...
+	/* Calculate elapsed CIG intervals until the instant */
+	cig_interval_us = cig->iso_interval * ISO_INT_UNIT_US;
+	latency_cig = DIV_ROUND_UP(elapsed_acl_us, cig_interval_us);
+	elapsed_cig_us = latency_cig * cig_interval_us;
+
+	/* Compensate for the difference between ACL elapsed vs CIG elapsed */
+	offset_min_us += elapsed_cig_us - elapsed_acl_us;
+	while (offset_min_us > (cig_interval_us + 500U)) {
+		offset_min_us -= cig_interval_us;
+	}
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/ull_central_iso.c:971]: Where does the `500U` come from?

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_peripheral_iso.c (hunk 1 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_peripheral_iso.c:414]: bn_tx start at 1. Hence, it is pretty much this:

```suggestion
				cis_lll->sn += cis_lll->tx.bn - (bn_tx - 1U);
```

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_peripheral_iso.c:414]: There quite a bit of duplicate code in LLL implementation, as the development has been iterative. Hope to refactor all at once, when starting to cover conformance tests.

File: subsys/bluetooth/controller/ll_sw/ull_central_iso.c (hunk 2 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_central_iso.c:960]: Good question, will rework on this based on spec maximums

File: subsys/bluetooth/controller/ll_sw/ull_central_iso.c (hunk 3 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_central_iso.c:971]: This is the spec defined minimum value for CIS offset, i did not find another use of this value in the implementation. On second thought, I will have a define for this.

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_central_iso.c]: Added define.

File: subsys/bluetooth/controller/ll_sw/ull_central_iso.c (hunk 2 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_central_iso.c:964]: added code comments.

**Review by Thalley — APPROVED** [review]

**Review by kruithofa — APPROVED** [review]

**Pull request merged**